### ISSUE TITLE
Discrepancy in rFVD Score for UCF-101 Test Set

### ISSUE THREAD

**Issue by flying0712** [TencentARC/SEED-Voken#57]
Thank you very much for your hard work and open-source contribution! I am testing the model's performance on the UCF-101 dataset, specifically focusing on the video reconstruction metric rFVD of Open-MAGVIT2.

I used the provided codebase and the pre-trained model `ucf101_lfqgan_128_262144.ckpt`. The only modifications I made were to the file paths and data loading code to ensure that `evaluation_video.py` runs smoothly on my setup. I tested the model on the `testlist01.txt`  of the UCF-101 test set using the following command:

```bash
python evaluation_video.py \
--config_file "configs/Open-MAGVIT2/gpu/ucf101_lfqgan_128_L.yaml" \
--ckpt_path "pretrained/ucf101_lfqgan_128_262144.ckpt"
```

The configuration file `gpu/ucf101_lfqgan_128_L.yaml` was mostly kept as-is:

```yaml
model:
    # use_ema: False # both False and True were tested
    # image_pretrain_path: pretrained/imagenet_128_L.ckpt # both on and off were tested

validation:
  target: src.Open_MAGVIT2.data.ucf101.VideoDataset
  params:
    config:
      data_folder: /data1/home/dataset/UCF-101
      size: 128
      mode: validation
      sequence_length: 17
      sample_every_n_frames: 1
      frame_sample_rate: 4
      subset: testlist01 # 3783 test videos
```

To ensure proper loading of test data, I slightly adjusted the data loading code in `ucf-101.py` as follows:  

```python
if subset is not None:
    # Original code: with open("../../data/ucf-101_{}.txt".format(subset), "r") as f:
    with open("/data1/home/dataset/ucfTrainTestlist/{}.txt".format(subset), "r") as f:
        # Original code: self.video_files = f.read().splitlines()
        self.video_files = [os.path.join(data_folder, line.split()[0]) for line in f.read().splitlines()]
```

After running multiple tests, I consistently obtained an FVD score around 30, which is significantly **higher than the value of 18 reported in the paper**. Here are some sample results:

- **Test 1:**
  - FVD: **30**.676300048828125
  - LPIPS_ALEX: 0.05063600838184357
  - LPIPS_VGG: 0.09647740423679352
  - 
- **Test 2:**
  - FVD: **33**.65459060668945
  - LPIPS_ALEX: 0.050794098526239395
  - LPIPS_VGG: 0.09628558158874512
  - Utilization: 0.9995002746582031

- **Test 3:**
   - FVD:  **32**.173744201660156
   - LPIPS_ALEX:  0.050947342067956924
   - LPIPS_VGG:  0.09680213034152985
   - utilization 0.9994964599609375

I am wondering if there might be an issue with my testing methodology or if there are differences between my setup and the one used for the original evaluation. Is it reasonable to see such a discrepancy in the rFVD score?

Additionally, I noticed that **the rFVD is significantly affected by the size of the test dataset**: the larger the test dataset, the smaller the rFVD. For example, when the test dataset size is **50, 200, and 3783,** the rFVD values are approximately **430, 180, and 30**, respectively. Is this behavior expected?

Thank you very much for your assistance!

**Comment by RobertLuo1** [comment]
Hi, Thanks for your attention to our work. Actually, the rFID is highly correlated to the scale of the dataset and more samples ensure more stable results. As shown in the [dataset](https://github.com/TencentARC/SEED-Voken/blob/ec8ad8e9b71704f9e0ba14ff699f1733cb7dc5e8/src/Open_MAGVIT2/data/ucf101.py#L53), we follow the evaluation protocol adopted in the original magvit-v2 paper, which tests on the UCF-101 training sets with 10000 videos. I suppose that the aligned evaluation setting plays a significant role in reproducing the final results.

**Comment by flying0712** [comment]
Thank you very much for your prompt response! Following your suggestion, I tested the model on the training set with over 9500 videos and obtained an rFVD of around 19, which is much closer to the value reported in the paper.  Thank you again for your guidance!
